File: BOJ/q8448.cpp
#include <iostream>

typedef long long ll;
const ll INF = 1e12;
const int LEN = 5e5;

int N;
ll pivot[8];
ll p[LEN][3];

int main() {
    std::cin.tie(0)->sync_with_stdio(0);
    std::cin >> N;
    for (int i = 0; i < 8; ++i) pivot[i] = -INF;
    for (int i = 0; i < N; ++i) {
        std::cin >> p[i][0] >> p[i][1] >> p[i][2];
        for (int b = 0; b < 8; ++b) {
            ll cur = 0;
            for (int j = 0; j < 3; ++j) {
                if (b & 1 << j) cur += p[i][j];
                else cur -= p[i][j];
            }
            pivot[b] = std::max(pivot[b], cur);
        }
    }
    for (int i = 0; i < N; ++i) {
        ll ret = 0;
        for (int b = 0; b < 8; ++b) {
            ll cur = pivot[b];
            for (int j = 0; j < 3; ++j) {
                if (~b & 1 << j) cur += p[i][j];
                else cur -= p[i][j];
            }
            ret = std::max(ret, cur);
        }
        std::cout << ret << '\n';
    }
}

File: BOJ/q8448.rs
use std::io::{self, BufRead, Write};
use std::cmp::max;

const INF: i64 = 1e12 as i64;

fn main() {
    let mut iter = io::stdin().lock().lines();
    let mut writer = io::BufWriter::new(io::stdout());
    let n: usize = iter.next().unwrap().unwrap().trim().parse().unwrap();
    let mut pivot = [-INF; 8];
    let mut pos: Vec<Vec<i64>> = vec![];
    for _ in 0..n {
        let p: Vec<i64> = iter.next().unwrap().unwrap()
            .split_whitespace().map(|x| x.parse().unwrap()).collect();
        for i in 0..8 {
            let mut x = 0;
            for j in 0..3 {
                if i & 1 << j == 0 {
                    x += p[j];
                } else {
                    x -= p[j];
                }
            }
            pivot[i] = max(pivot[i], x);
        }
        pos.push(p);
    }
    for k in 0..n {
        let mut ret = -INF;
        for i in 0..8 {
            let mut x = pivot[i];
            for j in 0..3 {
                if i & 1 << j == 0 {
                    x -= pos[k][j];
                } else {
                    x += pos[k][j];
                }
            }
            ret = max(ret, x);
        }
        _ = writeln!(writer, "{}", ret);
    }
}